pkg/kv/kvserver/replica_backpressure.go

```diff
@@ -108,6 +108,19 @@ func canBackpressureBatch(ba *roachpb.BatchRequest) bool {
 	return false
 }
 
+// canBypassReplicaCircuitBreakerForBatch returns whether the provided
+// BatchRequest bypasses the per-Replica circuit breaker. This is the
+// case if any request in the batch is requesting to do so.
+func canBypassReplicaCircuitBreakerForBatch(ba *roachpb.BatchRequest) bool {
```

Comment:
I think this should be a method on `BatchRequest`, we already have helpers there like `hasFlag()` to do this.

Also, would it make sense to require `isAlone` in `flagDependencies` for this? I think that's already the case for the requests we're flagging. Then the DistSender would automatically split them out anyway.

Reply:
I'll keep this where it is for now, it follows the precedent set by `canBackpressureBatch`.
Interesting idea about `isAlone`, but I don't see why a request that wants to bypass circuit breakers needs to be alone. For example, one might do multiple `AddSSTable` calls in one batch. We don't do that today, but it seems like an artificial constraint. If anything, we could teach `DistSender` to also split along the new flag, but this strikes me as premature.